protocols/v2/framing-sv2/src/framing2.rs

```diff
@@ -31,31 +33,35 @@ pub trait Frame<'a, T: Serialize + GetSize>: Sized {
     type Buffer: AsMut<[u8]>;
     type Deserialized;
 
-    /// Serialize the frame into dst if the frame is already serialized it just swap dst with
-    /// itself
+    /// Write the serialized `Frame` into `dst`.
     fn serialize(self, dst: &mut [u8]) -> Result<(), Error>;
 
-    //fn deserialize(&'a mut self) -> Result<Self::Deserialized, serde_sv2::Error>;
-
+    /// Get the payload
     fn payload(&'a mut self) -> &'a mut [u8];
 
-    /// If is an Sv2 frame return the Some(header) if it is a noise frame return None
+    /// `Sv2Frame` returns `Some(self.header)`, `NoiseFrame` return `None`.
```

Comment:
Since NoiseFrame is no more a thing I would at least change the name everywhere and use HandShakeFrame instead. Here we can say return a frame when the frame have an header like Sv2Frame, when it do not (like HandShakeFrame) return None.